cleanrepo/CleanRepo/Program.cs

```diff
@@ -1299,6 +1253,66 @@ private static void ListPopularFiles(List<FileInfo> tocFiles, List<FileInfo> mar
     #endregion
 
     #region Generic helper methods
+    /// <summary>
+    /// Gets the actual (case-sensitive) file path on the file system for a specified case-insensitive file path.
+    /// </summary>
+    private static string GetActualCaseForFilePath(string pathAndFileName)
+    {
+        string directory = Path.GetDirectoryName(pathAndFileName);
+        string directoryCaseSensitive = GetDirectoryCaseSensitive(directory);
+        string pattern = $"{Path.GetFileName(pathAndFileName)}.*";
+        string resultFileName;
+
+        if (directoryCaseSensitive is null)
+            throw new FileNotFoundException($"File not found: {pathAndFileName}.", pathAndFileName);
+
+        // Enumerate all files in the directory, using the file name as a pattern.
+        // This lists all case variants of the filename, even on file systems that are case sensitive.
+        IEnumerable<string> foundFiles = Directory.EnumerateFiles(
+            directoryCaseSensitive, 
+            pattern, 
+            new EnumerationOptions { MatchCasing = MatchCasing.CaseInsensitive });
+
+        if (foundFiles.Any())
+        {
+            resultFileName = foundFiles.First();
+        }
+        else
+        {
+            throw new FileNotFoundException($"File not found: {pathAndFileName}.", pathAndFileName);
+        }
```

Comment:
nit: I'm seeing both styles of single line if/else, those with brackets and those without. I prefer without :)